holidays/long_weekends_finder.py

```diff
@@ -0,0 +1,141 @@
+            - end_date (date): The end date of the long weekend
+            - duration (int): The number of days in the long weekend
+            - holidays (list): The names of holidays included in the long weekend
+
+    Raises:
+        ValueError: If year, month, or country is invalid.
+    """
+
+    # Validate country
+    try:
+        holiday_obj = holidays.country_holidays(country, years=year, language=language)
+    except Exception as e:
+        raise ValueError(f"Error fetching holidays: {e}")
+
+    weekends = holiday_obj.weekend
+    working_weekend_days = getattr(holiday_obj, "weekend_workdays", set())
+
+    # Extend holidays to cover the last week of the previous year and first week of the next year
+    if month is None and not exact_range:
+        prev_year_holidays = {
+            d: name
+            for d, name in holidays.country_holidays(
+                country, years=year, language=language
+            ).items()
+            if d.month == 12 and d.day >= 25  # Last week of December
+        }
+        next_year_holidays = {
+            d: name
+            for d, name in holidays.country_holidays(
+                country, years=year + 1, language=language
+            ).items()
+            if d.month == 1 and d.day <= 7  # First week of January
+        }
+        holidays_filtered = {**prev_year_holidays, **holiday_obj, **next_year_holidays}
+    else:
+        holidays_filtered = {d: name for d, name in holiday_obj.items()}
+
+    if month is not None:
+        first_day_of_month = date(year, month, 1)
+        last_day_of_month = date(year, month, calendar.monthrange(year, month)[1])
+
+        # Allow spillovers only when exact_range is False
+        if not exact_range:
+            first_day_of_month -= timedelta(days=2)
+            last_day_of_month += timedelta(days=2)
+
+        holidays_filtered = {
+            d: name
+            for d, name in holidays_filtered.items()
+            if first_day_of_month <= d <= last_day_of_month
+        }
+
+    long_weekends = []
+    seen_dates = set()
+
+    for holiday_date, holiday_name in holidays_filtered.items():
+        if holiday_date in seen_dates:
+            continue
+
+        start_date, end_date = holiday_date, holiday_date
+        holidays_involved = [holiday_name]
+
+        prev_day = start_date - timedelta(days=1)
+        while prev_day in holidays_filtered or (
+            prev_day.weekday() in weekends and prev_day not in working_weekend_days
+        ):
+            if prev_day in holidays_filtered:
+                holidays_involved.insert(0, holidays_filtered[prev_day])
+            start_date = prev_day
+            seen_dates.add(prev_day)
+            prev_day -= timedelta(days=1)
+
+        next_day = end_date + timedelta(days=1)
+        while next_day in holidays_filtered or (
+            next_day.weekday() in weekends and next_day.weekday() not in working_weekend_days
```

Comment:
I have already pointed out this [here](https://github.com/vacanza/holidays/pull/2270#discussion_r1965885548).